### ISSUE TITLE
A couple of questions

### ISSUE THREAD

**Issue by DataWookie** [YettyILU/Home-Credit-Default-Risk#1]
Hi Yetunde,

Thanks for this. Your submission looks good. I'm impressed that you managed to put together such a complete and elegant solution so quickly. Can you give me an idea of your workflow?

Also I see that you apply feature selection. I'm always in two minds about doing this with a tree based model. My feeling is that the trees perform implicit feature selection, so my reducing the feature space before building the trees you are actually reducing the potential of the model. What are your thoughts?

Thanks,
Andrew.

**Comment by YettyILU** [comment]
I agreed with you that tree based model perform implicit feature selection. I applied feature selection before model for the following reasons:
1. To reduce the dimensionality of the data by eliminating features that are less relevant to the problem.
2. To help reduce the risk of overfitting.
3. To help reduce the computational time.
NOTE: I have tried this in practice and discover that it has little or no effect on the overall performance of the model.

**Comment by YettyILU** [comment]
Good morning Andrew,
The step 2_datapreprocessing was from kernels in Kaggle while every other thing is my work.
ThanksYetunde

Sent from Yahoo Mail on Android 
 
  On Wed, Aug 8, 2018 at 8:21 PM, Andrew Collier<notifications@github.com> wrote:   
Hi Yetunde,

Thanks for answering my questions and also splitting up your analysis to kind of show your work flow.

I guess I have one final question and that is: what portion of your submission was original work and what was borrowed from elsewhere?

Best regards, Andrew.

**Comment by DataWookie** [comment]
Oh, okay. So am I right in understanding that the majority of your submission was derived from an existing Kaggle kernel? That's a little disappointing, don't you think? Why did you take this approach rather than working through it yourself?

**Comment by YettyILU** [comment]
Hi Andrew,
I hope you have a great weekend.
I learnt that the kaggle kernel is to assist one in solving a problem.For clarity, I said that only the pre-process step is from kernel not all my submission. There is nothing that is in the pre-process that I cannot do, although most of the time I like to do it in SQL to avoid bringing too large dataset into Rstudio.
I fitted different xgboost models, I only submitted the best model to you.1.I fitted model having the same parameters like the example I sent to you in my work.2.I imputed for missing values and filled xgboost with caret using random search for parameter turning.3.For variable selection, I used the Bourta package  in the example of my work while in the test assessment I used EFS package because Bourta  package cannot handle missing values.
As a person, I am after result and my research experience have been helping me to achieve speedy and accurate outcomes. As long as my outcomes are good and correct, I do not need to prove I am a "superman" by developing what is available and free to be used and will produce exactly the same results. I do not know your expectation in the type of a person you want to hire, but where I work, I am appreciated, I have learnt and have imparted others and I am still learning.
In conclusion, majority of my submission and effort comes from my experience.
Regards,Yetunde Ilupeju
 

    On Friday, 10 August 2018, 16:02:49 GMT+2, Andrew Collier <notifications@github.com> wrote:  
 
 
Oh, okay. So am I right in understanding that the majority of your submission was derived from an existing Kaggle kernel? That's a little disappointing, don't you think? Why did you take this approach rather than working through it yourself?

**Comment by DataWookie** [comment]
Hi Yetunde,

You don't need to defend yourself and I'm not looking for a superman/woman.

I am sure that you are appreciated where you currently work and, as I mentioned previously, I am really impressed with your submission.

However, as a potential new employer I am simply doing my due diligence and a part of that is asking questions. The objective being twofold:

(1) To get a better understanding of your submission (there was no attribution in your submission for the pre-processing that you got from the other kernel, it just happened to look familiar to me). Perhaps I am old fashioned, but I expected the submission to be original or at least to explicitly indicate what portion was copied from elsewhere. I am completely in favour of "standing on the shoulders of giants" but I also think that you understand the objective with this assessment was to see what *you* can do.
(2) To see how you respond to questions about your work.

Regards,
Andrew.

**Comment by YettyILU** [comment]
Good day Andrew,

Thanks for your response!

I like the way in which you are going about your hiring process, as it's important to understand my skills and the way in which I work. The way in which you ask questions is one of the things I admire about you.

It seems like there was a miscommunication. I was unaware that this assignment needed to be 100% my original work. I would like to let you know that if I was aware of this, I would have been happy to do so.

The dataset has 200 variables and in the pre_process step additional 29 variables was created, this is the reason why I used it. I felt that by telling you the truth, it would make you discover that I have a good research skill.

I believe that in working relationships, trust is very important. Especially with remote work.

We started very well, I wish we end it also well.

I am still very happy to move forward if you are still interested. 

Thanks
Yetunde
